src/field_weather.c

```diff
@@ -807,22 +830,53 @@ void UpdateSpritePaletteWithWeather(u8 spritePaletteIndex)
     default:
         if (gWeatherPtr->currWeather != WEATHER_FOG_HORIZONTAL)
         {
-            ApplyColorMap(paletteIndex, 1, gWeatherPtr->colorMapIndex);
+            if (gWeatherPtr->colorMapIndex)
+                ApplyColorMap(paletteIndex, 1, gWeatherPtr->colorMapIndex);
+            else
+                UpdateSpritePaletteWithTime(spritePaletteIndex);
         }
         else
         {
-            paletteIndex = PLTT_ID(paletteIndex);
-            BlendPalette(paletteIndex, 16, 12, RGB(28, 31, 28));
+            // In horizontal fog, only specific palettes should be fog-blended
+            if (allowFog)
+            {
+                i = min((gTimeOfDay + 1) * 4, 12); // fog coeff, highest in day and lowest at night
+                paletteIndex = PLTT_ID(paletteIndex);
+                // First blend with time
+                CpuFastCopy(gPlttBufferUnfaded + paletteIndex, gPlttBufferFaded + paletteIndex, PLTT_SIZE_4BPP);
+                UpdateSpritePaletteWithTime(spritePaletteIndex);
+                // Then blend faded->faded with fog coeff
+                BlendPalettesFine(1, gPlttBufferFaded + paletteIndex, gPlttBufferFaded + paletteIndex, i, RGB(28, 31, 28));
+            }
+            else
+            {
+                // Otherwise, just time-blend the palette
+                UpdateSpritePaletteWithTime(spritePaletteIndex);
+            }
         }
         break;
     }
+    // If faded out, i.e due to fadescreenswapbuffers,
+    // Copy unfaded palette to pal decomp buffer
+    // so it will be restored on fade-in
+    if (gPaletteFade.y == 16)
+        CpuFastCopy(
+            gPlttBufferUnfaded + OBJ_PLTT_ID(spritePaletteIndex),
+            gDecompressionBuffer + 2*OBJ_PLTT_ID(spritePaletteIndex),
+            PLTT_SIZE_4BPP
+        );
 }
 
-void ApplyWeatherColorMapToPal(u8 paletteIndex)
+void ApplyWeatherColorMapToPal(u8 paletteIndex) // now unused / obselete
```

Comment:
If unused, we should either mark it as `UNUSED` or (imo) outright remove it

Reply:
Done